### PR TITLE
Change hour of daily tasks to run at 1am and only run full backups on weekends

### PR DESCRIPTION
**Pull request by zoof**
With large amounts of email, full backups can take a lot of time.  This PR addresses the first option suggested from https://discourse.mailinabox.email/t/full-backups-taking-a-long-time/12242.

### CODE REVIEW

**Comment by kiekerjan** [comment]
I don't think it works like that. The variable full_backup is only set to True when backup.py is called with "--full". You should look at should_force_full

File: management/backup.py [hunk 1]
```diff
@@ -276,11 +280,11 @@ def perform_backup(full_backup):
...
 	# On the first run, always do a full backup. Incremental
-	# will fail. Otherwise do a full backup when the size of
-	# the increments since the most recent full backup are
-	# large.
+	# will fail. Otherwise do a full backup on weekends when
+	# the size of the increments since the most recent full
+	# backup are large.
 	try:
-		full_backup = full_backup or should_force_full(config, env)
+		full_backup = (full_backup and weekend) or should_force_full(config, env)
```
Comment (JoshData) [inline management/backup.py:287]: I think @kiekerjan is right, so it should probably be:

```suggestion
		full_backup = full_backup or (weekend and should_force_full(config, env))
```

By doing it `weekend and ...`, if `weekend` is false it won't even check if it should (via `should_force_full`). `should_force_full` scans the existing backup to see it's state and may be somewhat slow, so avoiding it on non-weekend days would be nice.

Comment (kiekerjan) [reply management/backup.py:287]: The thing is that should_force_full also checks if there are no full backups. If in that case weekend = False, the backup will fail, as an incremental backup will be tried.

Comment (zoof) [reply management/backup.py]: Ok, I made changes to `should_force_full` instead.

**Comment by JoshData** [comment]
I tried to simplify it and then re-read the comments and realized why it was like that. All good. Thanks!

**Pull request merged**